keras_cv/losses/focal_test.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.losses.focal import FocalLoss
+
+
+class FocalTest(tf.test.TestCase):
+    def test_output_shape(self):
+        y_true = tf.cast(
+            tf.random.uniform(shape=[2, 5], minval=0, maxval=2, dtype=tf.int32),
+            tf.float32,
+        )
+        y_pred = tf.random.uniform(shape=[2, 5], minval=0, maxval=1, dtype=tf.float32)
+
+        focal_loss = FocalLoss()
+
+        self.assertAllEqual(focal_loss(y_true, y_pred).shape, [])
```

Comment:
you'll need to update the target shape to be [2]

Reply:
When reduction is auto, it aggregates over the dimensions. It could be a 4D array and the output will still have a shape of [].

Reply:
If you prefer keeping reduction="none" by default, I can do that. It may go against the other Keras losses though.

Reply:
my bad, sorry this comment is outdated